feat: permit2 bridging

lukasrosario opened the pull request:
adds permit2 bridging functionality

Commit:
feat(indexer): Add metrics to indexer/api

Commit:
feat(indexer): Connect api metrics to cli

Commit:
feat(indexer): add recoverer middleware

Commit:
refactors

refcell committed:
Add Game Status Tracking GaugeVec

Spacesai1or committed:
Init standard UFM Test Service architecture

Spacesai1or committed:
Init standard UFM Test Service architecture

Spacesai1or committed:
Init MetaMask Test Service

Spacesai1or committed:
Init MetaMask Test Service

Commit:
:recycle: chore: Make port configurable

:bug: fix: Fix port env var

Commit:
:wrench: chore: update schema.prisma

refcell committed:
Track in flight games inside the scheduler component.

hamdiallam committed:
Legacy block processor

Spacesai1or committed:
Init synpress and playwright

ajsutton committed:
Add initial documentation for fault proof alpha.

ajsutton committed:
op-e2e: Move geth wait utils to e2eutils

ajsutton committed:
op-e2e: Move geth init utils to e2eutils

ImTei committed:
specs: update span-batch format and rules

- Update span-batch format
- Fix batch derivation-rules
- Add optimization-stategies section

mergify[bot] committed:
Merge branch 'develop' into aj/recreate-dir

OptimismBot committed:
Merge pull request #7097 from ethereum-optimism/aj/recreate-dir

op-program: Recreate preimage data dir if required

mergify[bot] committed:
Merge branch 'develop' into aj/require-e2e

OptimismBot committed:
Merge pull request #7085 from ethereum-optimism/aj/require-e2e

ci: Add additional CI tasks that weren't previously required as bedrock-go-tests dependencies

mergify[bot] committed:
Merge branch 'develop' into span-batch-spec-fixes

OptimismBot committed:
Merge pull request #6363 from ethereum-optimism/span-batch-spec-fixes

specs: span-batch spec fixes

mergify[bot] committed:
Merge branch 'develop' into aj/faster-dockerfile

OptimismBot committed:
Merge pull request #7105 from ethereum-optimism/aj/faster-dockerfile

Dockerfiles: Run go mod download with just the go.mod and go.sum files

clabby committed:
Add `VMStatus` type

dedup

lukasrosario committed:
forge install: permit2

protolambda committed:
op-node: implement runtime config reloading

OptimismBot committed:
Merge pull request #7080 from ethereum-optimism/refcell/game-gauge

feat(op-challenger): Game GaugeVec

mergify[bot] committed:
Merge branch 'develop' into refcell/inflight-gauge

OptimismBot committed:
Merge pull request #7086 from ethereum-optimism/refcell/inflight-gauge

feat(op-challenger): Inflight game metrics

hamdiallam committed:
fix bridge processor height detection on startup

ajsutton committed:
op-e2e: Update e2e tests to demonstrate issues with vm status.

ajsutton committed:
Update FPA docs based on review feedback.

OptimismBot committed:
Merge pull request #7109 from ethereum-optimism/aj/e2e-utils

Move geth e2eutils to e2eutils/geth package

ajsutton committed:
Create an e2e test showing cannon proving an output root is invalid

Spacesai1or committed:
Init Metmask send tx test

roninjin10 committed:
lowercase

Commit:
:bug: bad rebase

OptimismBot committed:
Merge pull request #7038 from ethereum-optimism/08-28-feat_indexer_Add_metrics_to_indexer/api

feat(indexer): Add metrics to indexer/api

mergify[bot] committed:
Merge branch 'develop' into 08-28-feat_indexer_add_recoverer_middleware

OptimismBot committed:
Merge pull request #7040 from ethereum-optimism/08-28-feat_indexer_add_recoverer_middleware

feat(indexer): add recoverer middleware

mergify[bot] committed:
Merge branch 'develop' into willc/port

roninjin10 committed:
Merge pull request #7102 from ethereum-optimism/dependabot/npm_and_yarn/babel/eslint-parser-7.22.15

build(deps-dev): bump @babel/eslint-parser from 7.22.10 to 7.22.15

mergify[bot] committed:
Merge branch 'develop' into willc/port

dependabot[bot] committed:
build(deps-dev): bump @vitest/coverage-istanbul from 0.34.1 to 0.34.3

Bumps [@vitest/coverage-istanbul](https://github.com/vitest-dev/vitest/tree/HEAD/packages/coverage-istanbul) from 0.34.1 to 0.34.3.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v0.34.3/packages/coverage-istanbul)

---
updated-dependencies:
- dependency-name: "@vitest/coverage-istanbul"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7089 from ethereum-optimism/willc/port

:recycle: chore: Make port configurable

mergify[bot] committed:
Merge branch 'develop' into willc/update-prisma

OptimismBot committed:
Merge pull request #7090 from ethereum-optimism/willc/update-prisma

:wrench: chore: update schema.prisma

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/vitest/coverage-istanbul-0.34.3

lukasrosario committed:
add permit2 bridging

lukasrosario committed:
add permit2 address to goerli deploy config to fix test

hamdiallam committed:
Merge pull request #7112 from ethereum-optimism/indexer.bridge.starting.heights

fix(indexer): bridge processor detecting the right starting height

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/vitest/coverage-istanbul-0.34.3

OptimismBot committed:
Merge pull request #7021 from ethereum-optimism/dependabot/npm_and_yarn/vitest/coverage-istanbul-0.34.3

build(deps-dev): bump @vitest/coverage-istanbul from 0.34.1 to 0.34.3

mergify[bot] committed:
Merge branch 'develop' into aj/cannon-invalid-output-root

OptimismBot committed:
Merge pull request #7114 from ethereum-optimism/aj/cannon-invalid-output-root

Create an e2e test showing cannon proving an output root is invalid

mergify[bot] committed:
Merge branch 'develop' into aj/fpa-docs

maurelian committed:
feat(style): Document style for testing solidity getters

OptimismBot committed:
Merge pull request #7099 from ethereum-optimism/aj/fpa-docs

Add initial documentation for fault proof alpha

Inphi committed:
specs: Update dispute game interfaces

refcell committed:
Fixes small markdown header.

hamdiallam committed:
small bugfix

hamdiallam committed:
operators were right before

hamdiallam committed:
fix bridge startup

OptimismBot committed:
Merge pull request #7118 from ethereum-optimism/indexer.bridge.bugfix

fix(indexer): bridge l2 starting height &  l1/l2 logs

Inphi committed:
specs: Add Fault Dispute Game specs

dependabot[bot] committed:
build(deps): bump github.com/ethereum/go-ethereum in /op-exporter

Bumps [github.com/ethereum/go-ethereum](https://github.com/ethereum/go-ethereum) from 1.10.17 to 1.12.1.
- [Release notes](https://github.com/ethereum/go-ethereum/releases)
- [Commits](https://github.com/ethereum/go-ethereum/compare/v1.10.17...v1.12.1)

---
updated-dependencies:
- dependency-name: github.com/ethereum/go-ethereum
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

mergify[bot] committed:
Merge branch 'develop' into refcell/small-manual-header-nits

OptimismBot committed:
Merge pull request #7117 from ethereum-optimism/refcell/small-manual-header-nits

fix(docs): Challenger Doc Fixes

mergify[bot] committed:
Merge branch 'develop' into dependabot/go_modules/op-exporter/github.com/ethereum/go-ethereum-1.12.1

clabby committed:
Do not ignore status byte on VM step / in the chal solver

felipe-op committed:
add cooldown

felipe-op committed:
config update

felipe-op committed:
improve error metrics

felipe-op committed:
use network nonce

felipe-op committed:
use network gas

felipe-op committed:
remove gas configs

felipe-op committed:
send mutex

felipe-op committed:
add debug tools to docker file

felipe-op committed:
update example.config.toml

clabby committed:
Remove concept of valid / invalid from alphabet game

clabby committed:
Add error to `StateWitness::StateHash`

clabby committed:
:broom:

clabby committed:
Allow for a game to be created with a root claim committing to a paniced VM state

Fix test

clabby committed:
Add `TestStateHash`

ajsutton committed:
op-e2e: Remove duplicate logic for playing through a game

OptimismBot committed:
Merge pull request #7123 from ethereum-optimism/dependabot/go_modules/op-exporter/github.com/ethereum/go-ethereum-1.12.1

build(deps): bump github.com/ethereum/go-ethereum from 1.10.17 to 1.12.1 in /op-exporter

mergify[bot] committed:
Merge branch 'develop' into fpvm-status

dependabot[bot] committed:
build(deps): bump golang.org/x/term from 0.11.0 to 0.12.0

Bumps [golang.org/x/term](https://github.com/golang/term) from 0.11.0 to 0.12.0.
- [Commits](https://github.com/golang/term/compare/v0.11.0...v0.12.0)

---
updated-dependencies:
- dependency-name: golang.org/x/term
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/ethereum/go-ethereum in /op-ufm

Bumps [github.com/ethereum/go-ethereum](https://github.com/ethereum/go-ethereum) from 1.12.0 to 1.12.1.
- [Release notes](https://github.com/ethereum/go-ethereum/releases)
- [Commits](https://github.com/ethereum/go-ethereum/compare/v1.12.0...v1.12.1)

---
updated-dependencies:
- dependency-name: github.com/ethereum/go-ethereum
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #6913 from ethereum-optimism/fpvm-status

cannon,op-challenger,dispute-game: VM Status prefix byte in claim hash

protolambda committed:
op-node: update config-reloading comment

Co-authored-by: Adrian Sutton <adrian@oplabs.co>

protolambda committed:
op-e2e: use wait.ForReceiptOK helper instead of bind.WaitMined

mergify[bot] committed:
Merge branch 'develop' into runtime-config-reloading

OptimismBot committed:
Merge pull request #7111 from ethereum-optimism/runtime-config-reloading

op-node: implement runtime config reloading

anikaraghu commented on packages/contracts-bedrock/foundry.toml:
```diff
@@ -11,7 +11,9 @@ remappings = [
-  'ds-test/=lib/forge-std/lib/ds-test/src'
+  'ds-test/=lib/forge-std/lib/ds-test/src',
+  '@uniswap/permit2/=lib/permit2/src',
+  'solmate/=lib/solmate'
```
nit: can probably remove, seems this remapping is captured above and I dont see it being used anywhere

lukasrosario replied:
permit2 uses this remapping & compilation breaks unless we put it here too :/ . not ideal but foundry team is aware of this nested remapping issue

anikaraghu commented on packages/contracts-bedrock/deploy-config/goerli.json:
```diff
@@ -38,5 +38,6 @@
   "eip1559Elasticity": 10,
-  "systemConfigStartBlock": 8300214
+  "systemConfigStartBlock": 8300214,
+  "permit2Address": "0x000000000022D473030F116dDEE9F6B43aC78BA3"
```
do we need to add to mainnet config as well?

lukasrosario replied:
added mainnet & sepolia

anikaraghu commented on packages/contracts-bedrock/scripts/Deploy.s.sol:
```diff
@@ -428,7 +429,9 @@ contract Deploy is Deployer {
     function deployL1StandardBridge() public broadcast returns (address addr_) {
-        L1StandardBridge bridge = new L1StandardBridge();
+        L1StandardBridge bridge = new L1StandardBridge({
+            _permit2: IPermit2(cfg.permit2Address())
```
This looks good, just note that it this codepath is only hit when a full new L1 contracts deploy is done (which wouldn't be done again for chains that already exist, but would for new OP chains). Since our bridge contract is already deployed we would instead need to do a contract upgrade to store the permit2 address

lukasrosario replied:
👍 makes sense

anikaraghu commented on packages/contracts-bedrock/src/L1/L1StandardBridge.sol:
```diff
@@ -69,7 +70,10 @@ contract L1StandardBridge is StandardBridge, Semver {
     /// @notice Constructs the L1StandardBridge contract.
-    constructor() Semver(1, 2, 1) StandardBridge(StandardBridge(payable(Predeploys.L2_STANDARD_BRIDGE))) {
+    constructor(IPermit2 _permit2)
+        Semver(1, 2, 1)
```
we should bump the version. might even be a major version change https://semver.org/ since it changes the API (although it doesn't change existing functions that would be called, just adds additional ones, so maybe just a minor version change is fine)
```suggestion
        Semver(1, 3, 0)
```

lukasrosario replied:
updated

anikaraghu commented on packages/contracts-bedrock/src/L2/L2StandardBridge.sol:
```diff
@@ -54,7 +55,13 @@ contract L2StandardBridge is StandardBridge, Semver {
+        StandardBridge _otherBridge,
+        IPermit2 _permit2
+    )
+        Semver(1, 2, 1)
```
need to bump the version and we'll also have to regenerate bindings in `op-bindings`

anikaraghu commented on packages/contracts-bedrock/src/L1/L1StandardBridge.sol:
```diff
@@ -168,6 +172,66 @@ contract L1StandardBridge is StandardBridge, Semver {
         _initiateERC20Deposit(_l1Token, _l2Token, msg.sender, _to, _amount, _minGasLimit, _extraData);
     }
 
+    function depositERC20WithPermit2(
```
We should confirm with OP whether to keep the old functionality or by default use Permit2 for deposits. How you did it makes sense and is probably the correct way, but worth documenting this as a design decision we made

clabby committed:
Add `forge clean` to `op-bindings` Makefile

anikaraghu commented on packages/contracts-bedrock/src/universal/StandardBridge.sol:
```diff
@@ -51,6 +52,20 @@ abstract contract StandardBridge is Initializable {
     ///         would be a multiple of 50.
     uint256[46] private __gap;
 
+    IPermit2 public immutable permit2;
```
let's define this in the same place where the other immutable variables are defined above.

also it looks like Optimism recently made the `messenger` variable a state variable instead of an `immutable` variable - I looked into why and it's because that variable can be different for different OP chains, and they want to eventually be able to use the same proxy for all OP chains. Since Permit2 should be deployed at the same address for all chains, should be ok to use an immutable 👍

lukasrosario replied:
updated

mergify[bot] committed:
Merge branch 'develop' into aj/fpvm-status-e2e

OptimismBot committed:
Merge pull request #7134 from ethereum-optimism/aj/fpvm-status-e2e

op-e2e: Un-skip TestCannonChallengeWithCorrectRoot

mergify[bot] committed:
Merge branch 'develop' into clabby/bindings/forge-clean

anikaraghu commented on packages/contracts-bedrock/src/L1/L1StandardBridge.sol:
```diff
@@ -168,6 +172,66 @@ contract L1StandardBridge is StandardBridge, Semver {
+        _initiateERC20DepositWithPermit2(_signatureTransferData, _l2Token, msg.sender, _to, _minGasLimit, _extraData);
+    }
+
+    function batchDepositERC20WithPermit2(
```
Also worth confirming/discussing whether we want to support batch deposits initially

anikaraghu commented on packages/contracts-bedrock/src/L2/L2StandardBridge.sol:
```diff
@@ -54,7 +55,13 @@ contract L2StandardBridge is StandardBridge, Semver {
+    )
+        Semver(1, 2, 1)
+        StandardBridge(_otherBridge, _permit2)
+    {
```
are we planning to add permit2 to withdrawals as well?

lukasrosario replied:
i dont think it would be applicable to withdrawals because a withdrawal just burns the mintable token instead of doing an actual erc20 transfer

anikaraghu commented on packages/contracts-bedrock/src/universal/StandardBridge.sol:
```diff
@@ -352,6 +426,126 @@ abstract contract StandardBridge is Initializable {
+            for (uint256 i = 0; i < numPermitted; ++i) {
+                deposits[_signatureTransferData.permit.permitted[i].token][_remoteTokens[i]] = deposits[_signatureTransferData
+                    .permit
+                    .permitted[i].token][_remoteTokens[i]] + _signatureTransferData.permit.permitted[i].amount;
```
can we confirm this is ok to do in an unchecked loop? And if so, David suggested doing this in a second unchecked loop instead of one big unchecked loop. Should have the same effect but basically if we add some additional logic in the loop sometime in the future, we have a higher likelihood of doing this in the unchecked loop unintentionally when it has this large scope

lukasrosario replied:
you're right i actually don't think this line should be unchecked. updated

anikaraghu commented on packages/contracts-bedrock/src/universal/StandardBridge.sol:
```diff
@@ -352,6 +426,126 @@ abstract contract StandardBridge is Initializable {
+    {
+        uint256 numPermitted = _signatureTransferData.permit.permitted.length;
+
+        unchecked {
+            for (uint256 i = 0; i < numPermitted; ++i) {
```
Gas optimization opportunity (this was pointed out in a different audit we had with the following explanation): Under certain safe scenarios, for loops in Solidity can be optimized by taking advantage of
the following properties:
• Un-initialized unsigned integers are set to 0 by default.
• Caching variables that are read multiple times reduces gas costs as excess SLOADs are avoided this
way.
• Overflow protection can be disabled if the variable to increment has a clearly defined upper bound.
• The prefix increment is the cheapest way to increament a counter.

```suggestion
        for (uint256 i ; i < numPermitted; ) {
          ...
          unchecked { ++i; }
        }
```

lukasrosario replied:
good to know. i think all of the loops i added adhere to these but lmk if im missing something specific

anikaraghu commented on packages/contracts-bedrock/src/universal/StandardBridge.sol:
```diff
@@ -352,6 +426,126 @@ abstract contract StandardBridge is Initializable {
         // contracts may override this function in order to emit legacy events as well.
         _emitERC20BridgeInitiated(_localToken, _remoteToken, _from, _to, _amount, _extraData);
 
+        _sendFinalizeBridgeERC20Message(_localToken, _remoteToken, _from, _to, _amount, _extraData, _minGasLimit);
```
I like how we share the code here, wondering if we could share even more of the code between this and the new path? seems like there's still a decent amount of overlap

lukasrosario replied:
not sure if im interpreting this correctly but i also added a new `_increaseDeposits` and `_decreaseDeposits` to replace the `deposits[_localToken][_remoteToken] = deposits[_localToken][_remoteToken] + _amount;` that exists throughout

lukasrosario replied:
lmk if you think we should do something similar for anything else

ajsutton committed:
op-e2e: Add call to step that verifies it is reject when matching the original claim. (#7133)

mergify[bot] committed:
Merge branch 'develop' into clabby/bindings/forge-clean

clabby committed:
chore: :broom: `foundry.toml`

OptimismBot committed:
Merge pull request #7137 from ethereum-optimism/clabby/bindings/forge-clean

fix(op-bindings): Add `forge clean` to make's `compile` directive

mergify[bot] committed:
Merge branch 'develop' into clabby/ctb/clean-foundry-toml

OptimismBot committed:
Merge pull request #7138 from ethereum-optimism/clabby/ctb/clean-foundry-toml

chore: Clean up `foundry.toml`

Commit:
:bug: fix: update prisma schema

lbeder committed:
EAS v1.2.0: gas optimizations, minor improvements and fixes, and some typo and NATSPEC fixes (#7094)

* Add an index and the schema record data to the Registered event

* Gas Optimization: Cache resolver.isPayable() external call

* Add extra multiRequest.data.length == 0 validation

* Gas Gas Optimization: Cache Revoked event arguments

* Make sure to refund the attester/revoker in case of non-last attestation/revocation and where there is no resolver

* Gas Optimization: cache loop iteration lengths

* Gas Optimization: optimize _resolveAttestation() and _resolveAttestations()

* Fix missing Natspec, renaming variables, and incorrect comments

* Add explicit deadline to delegated attestation/revocation requests

* Provide an option for users to invalidate nonces by increasing their nonces to (higher) new values

* Add extra input length validation in multiRevoke and multiAttest functions

* Updates sample usage in comments

* Capture the payment amount in signatures

* Bump version

* Emit NonceIncreased event when users increase their nonces

* Make delegate attestation deadline inclusive + handle prevent nonce overflow due to user error

* Minor fix

* Update bindings

* Fix lint

* Lock semver

* Add an index and the schema record data to the Registered event

* Gas Optimization: Cache resolver.isPayable() external call

* Add extra multiRequest.data.length == 0 validation

* Gas Gas Optimization: Cache Revoked event arguments

* Make sure to refund the attester/revoker in case of non-last attestation/revocation and where there is no resolver

* Gas Optimization: cache loop iteration lengths

* Gas Optimization: optimize _resolveAttestation() and _resolveAttestations()

* Fix missing Natspec, renaming variables, and incorrect comments

* Add explicit deadline to delegated attestation/revocation requests

* Provide an option for users to invalidate nonces by increasing their nonces to (higher) new values

* Add extra input length validation in multiRevoke and multiAttest functions

* Updates sample usage in comments

* Capture the payment amount in signatures

* Bump version

* Emit NonceIncreased event when users increase their nonces

* Make delegate attestation deadline inclusive + handle prevent nonce overflow due to user error

* Minor fix

* Update bindings

* Fix lint

* Lock semver

lukasrosario committed:
undo op contract upgrades and temporarily use permit2 fork

mergify[bot] committed:
Merge branch 'develop' into update-prisma

OptimismBot committed:
Merge pull request #7139 from ethereum-optimism/update-prisma

:bug: fix: update prisma schema

mergify[bot] committed:
Merge branch 'develop' into aj/e2e-duplication

OptimismBot committed:
Merge pull request #7135 from ethereum-optimism/aj/e2e-duplication

op-e2e: Remove duplicate logic for playing through a game

lukasrosario commented on .gitmodules:
```diff
@@ -13,3 +13,6 @@
 [submodule "packages/contracts-bedrock/lib/forge-std"]
 	path = packages/contracts-bedrock/lib/forge-std
 	url = https://github.com/foundry-rs/forge-std
+[submodule "packages/contracts-bedrock/lib/permit2"]
+	path = packages/contracts-bedrock/lib/permit2
+	url = https://github.com/lukasrosario/permit2
```
temporarily pointing to fork on my branch until pragma downgrade gets upstreamed

lukasrosario replied:
pointing at uniswap again

ajsutton committed:
Merge branch 'develop' into cleanup-devnet-modified-file

felipe-op committed:
remove gas configs

felipe-op committed:
timeout metrics

Commit:
:sparkles: feat(indexer): Add preset starting block heights

dependabot[bot] committed:
build(deps): bump viem from 1.3.1 to 1.10.7

Bumps [viem](https://github.com/wagmi-dev/viem) from 1.3.1 to 1.10.7.
- [Release notes](https://github.com/wagmi-dev/viem/releases)
- [Changelog](https://github.com/wagmi-dev/viem/blob/main/CHANGELOG.md)
- [Commits](https://github.com/wagmi-dev/viem/compare/viem@1.3.1...viem@1.10.7)

---
updated-dependencies:
- dependency-name: viem
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Inphi committed:
specs: remove Attestation DisputeGame interface

OptimismBot committed:
Merge pull request #7142 from ethereum-optimism/dependabot/npm_and_yarn/viem-1.10.7

build(deps): bump viem from 1.3.1 to 1.10.7

dependabot[bot] committed:
build(deps-dev): bump typescript from 5.1.6 to 5.2.2

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 5.1.6 to 5.2.2.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v5.1.6...v5.2.2)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mergify[bot] committed:
Merge branch 'develop' into aj/cleanup-devnet-modified-file

OptimismBot committed:
Merge pull request #7140 from ethereum-optimism/aj/cleanup-devnet-modified-file

Fix dangling modified file on make devnet-up

ajsutton committed:
fpa: Add docs that point to the e2e test for invalid proposals

ajsutton committed:
Point to version of test with improved comments.

ajsutton committed:
op-e2e: Remove redundant fault proof tests

TestMultipleAlphabetGames is covered by TestMultipleCannonGames
TestResolveDisputeGame is covered by every other test since they all resolve games

Spacesai1or committed:
Metamask send tx test Docker support

mergify[bot] committed:
Merge branch 'develop' into felipe/cooldown-b

OptimismBot committed:
Merge pull request #7132 from ethereum-optimism/felipe/cooldown-b

feat(ufm): add cooldown, always use network gas and nonce, better error metrics

mergify[bot] committed:
Merge branch 'develop' into aj/e2e-remove-dupe-tests

OptimismBot committed:
Merge pull request #7145 from ethereum-optimism/aj/e2e-remove-dupe-tests

op-e2e: Remove redundant fault proof tests

mergify[bot] committed:
Merge branch 'develop' into aj/invalid-root-docs

OptimismBot committed:
Merge pull request #7144 from ethereum-optimism/aj/invalid-root-docs

fpa: Add docs that point to the e2e test for invalid proposals

mergify[bot] committed:
Merge branch 'develop' into inphi/dispute-specs

OptimismBot committed:
Merge pull request #7122 from ethereum-optimism/inphi/dispute-specs

specs: Update Dispute Game Specifications

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/typescript-5.2.2

OptimismBot committed:
Merge pull request #7143 from ethereum-optimism/dependabot/npm_and_yarn/typescript-5.2.2

build(deps-dev): bump typescript from 5.1.6 to 5.2.2

dependabot[bot] committed:
build(deps-dev): bump mkdirp from 1.0.4 to 3.0.1

Bumps [mkdirp](https://github.com/isaacs/node-mkdirp) from 1.0.4 to 3.0.1.
- [Changelog](https://github.com/isaacs/node-mkdirp/blob/main/CHANGELOG.md)
- [Commits](https://github.com/isaacs/node-mkdirp/compare/v1.0.4...v3.0.1)

---
updated-dependencies:
- dependency-name: mkdirp
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7130 from ethereum-optimism/dependabot/npm_and_yarn/mkdirp-3.0.1

build(deps-dev): bump mkdirp from 1.0.4 to 3.0.1

mergify[bot] committed:
Merge branch 'develop' into wyatt/ufm/init

OptimismBot committed:
Merge pull request #7148 from ethereum-optimism/wyatt/ufm/init

Init standard UFM Test Service architecture

lukasrosario committed:
add permit2 deployment

mergify[bot] committed:
Merge branch 'develop' into wyatt/ufm/metamask-init

OptimismBot committed:
Merge pull request #7149 from ethereum-optimism/wyatt/ufm/metamask-init

Init MetaMask Test Service

mergify[bot] committed:
Merge branch 'develop' into jm/getter-test-style

dependabot[bot] committed:
build(deps-dev): bump nx from 16.7.3 to 16.8.1

Bumps [nx](https://github.com/nrwl/nx/tree/HEAD/packages/nx) from 16.7.3 to 16.8.1.
- [Release notes](https://github.com/nrwl/nx/releases)
- [Commits](https://github.com/nrwl/nx/commits/16.8.1/packages/nx)

---
updated-dependencies:
- dependency-name: nx
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7120 from ethereum-optimism/jm/getter-test-style

feat(style): Document style for testing solidity getters

lukasrosario committed:
switch back to uniswap permit2

anikaraghu commented on packages/contracts-bedrock/deploy-config/devnetL1.json:
```diff
@@ -41,10 +41,11 @@
-  "systemConfigStartBlock": 0
-}
+  "systemConfigStartBlock": 0,
+  "permit2Address": "0x000000000022D473030F116dDEE9F6B43aC78BA3"
```
instead of adding to the deploy-config, should we just inherit from the [constant](https://github.com/Uniswap/permit2/blob/79658f604fa017635177802a9e833b9065a13053/src/libraries/Permit2Lib.sol#L27) defined in Uniswap's code? since we assume Permit2 will always be deployed to the same addres

lukasrosario replied:
updated after adding permit2 to devnet at deterministic address

hamdiallam committed:
Fix bridge processor starting height & performance

hamdiallam committed:
Update indexer/config/config.go

Co-authored-by: Will Cory <willcory10@gmail.com>

mergify[bot] committed:
Merge branch 'develop' into preset-block-heights

OptimismBot committed:
Merge pull request #7141 from ethereum-optimism/preset-block-heights

:sparkles: feat(indexer): Add preset starting block heights

Spacesai1or committed:
Merge branch 'develop' into wyatt/ufm/synpress-playwright-init

OptimismBot committed:
Merge pull request #7150 from ethereum-optimism/wyatt/ufm/synpress-playwright-init

Init synpress and playwright

mergify[bot] committed:
Merge branch 'develop' into wyatt/ufm/metamask-send-tx-test

OptimismBot committed:
Merge pull request #7151 from ethereum-optimism/wyatt/ufm/metamask-send-tx-test

Init Metmask send tx test

mergify[bot] committed:
Merge branch 'develop' into wyatt/ufm/metamask-send-tx-test-docker

OptimismBot committed:
Merge pull request #7152 from ethereum-optimism/wyatt/ufm/metamask-send-tx-test-docker

Metamask send tx test Docker support

Commit:
:wrench: feat: Make indexer preset configurable in default indexer.toml

OptimismBot committed:
Merge pull request #7160 from ethereum-optimism/09-08-feat_Make_indexer_preset_configurable_in_default_indexer.toml

:wrench: feat: Make indexer preset configurable in default indexer.toml

hamdiallam committed:
indexer.db.batching

hamdiallam committed:
fire off bridge on startup. group large block ranges. Start right L1 Height

hamdiallam committed:
improve logging

hamdiallam committed:
improve logging

hamdiallam committed:
use the right ctc goerli contract

OptimismBot committed:
Merge pull request #7157 from ethereum-optimism/dependabot/npm_and_yarn/nx-16.8.1

build(deps-dev): bump nx from 16.7.3 to 16.8.1

dependabot[bot] committed:
build(deps-dev): bump zod from 3.22.0 to 3.22.2

Bumps [zod](https://github.com/colinhacks/zod) from 3.22.0 to 3.22.2.
- [Release notes](https://github.com/colinhacks/zod/releases)
- [Changelog](https://github.com/colinhacks/zod/blob/master/CHANGELOG.md)
- [Commits](https://github.com/colinhacks/zod/compare/v3.22.0...v3.22.2)

---
updated-dependencies:
- dependency-name: zod
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

tynes committed:
deps: update

mergify[bot] committed:
Merge branch 'develop' into dependabot/go_modules/golang.org/x/term-0.12.0

tynes committed:
deps: update hardhat

Updates the hardhat version so that a more recent version of node.js
can be used without warning. We should consider updating the node.js
version to the LTS version of node.js. The version of hardhat that
was being used was only supported by an older version of node.js

tynes committed:
Merge pull request #7162 from ethereum-optimism/deps/newer-hardhat

deps: update hardhat

tynes committed:
Merge pull request #7011 from ethereum-optimism/dependabot/npm_and_yarn/zod-3.22.2

build(deps-dev): bump zod from 3.22.0 to 3.22.2

tynes committed:
Merge pull request #7126 from ethereum-optimism/dependabot/go_modules/op-ufm/github.com/ethereum/go-ethereum-1.12.1

build(deps): bump github.com/ethereum/go-ethereum from 1.12.0 to 1.12.1 in /op-ufm

mergify[bot] committed:
Merge branch 'develop' into dependabot/go_modules/golang.org/x/term-0.12.0

tynes committed:
contracts-bedrock: use create2 to deploy implementations

Use create2 to deploy L1 contract implementations. This is useful
to deduplicate config, making the contracts have canoncial addresses
per version across networks. Every L1 contract implementation should
use the same create2 salt over time.

dependabot[bot] committed:
build(deps): bump github.com/supranational/blst in /op-ufm

Bumps [github.com/supranational/blst](https://github.com/supranational/blst) from 0.3.11-0.20230406105308-e9dfc5ee724b to 0.3.11.
- [Release notes](https://github.com/supranational/blst/releases)
- [Commits](https://github.com/supranational/blst/commits/v0.3.11)

---
updated-dependencies:
- dependency-name: github.com/supranational/blst
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/ethereum/go-ethereum in /proxyd

Bumps [github.com/ethereum/go-ethereum](https://github.com/ethereum/go-ethereum) from 1.12.0 to 1.12.1.
- [Release notes](https://github.com/ethereum/go-ethereum/releases)
- [Commits](https://github.com/ethereum/go-ethereum/compare/v1.12.0...v1.12.1)

---
updated-dependencies:
- dependency-name: github.com/ethereum/go-ethereum
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump chai and @types/chai

Bumps [chai](https://github.com/chaijs/chai) and [@types/chai](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/chai). These dependencies needed to be updated together.

Updates `chai` from 4.3.4 to 4.3.8
- [Release notes](https://github.com/chaijs/chai/releases)
- [Changelog](https://github.com/chaijs/chai/blob/4.x.x/History.md)
- [Commits](https://github.com/chaijs/chai/compare/v4.3.4...v4.3.8)

Updates `@types/chai` from 4.2.21 to 4.3.6
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/chai)

---
updated-dependencies:
- dependency-name: chai
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: "@types/chai"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7100 from ethereum-optimism/dependabot/go_modules/golang.org/x/term-0.12.0

build(deps): bump golang.org/x/term from 0.11.0 to 0.12.0

mergify[bot] committed:
Merge branch 'develop' into indexer.db.batching

dependabot[bot] committed:
build(deps): bump golang.org/x/crypto from 0.12.0 to 0.13.0

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.12.0 to 0.13.0.
- [Commits](https://github.com/golang/crypto/compare/v0.12.0...v0.13.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

OptimismBot committed:
Merge pull request #7156 from ethereum-optimism/indexer.db.batching

fix(indexer): batch database inserts and add indexes

mergify[bot] committed:
Merge branch 'develop' into indexer.bridge.interval

OptimismBot committed:
Merge pull request #7161 from ethereum-optimism/indexer.bridge.interval

fix(indexer): run bridge processor on startup & split large unobserved epochs

mergify[bot] committed:
Merge branch 'develop' into indexer.bridge.l2.start

tynes committed:
contracts-bedrock: migrate to ISemver

Migrate away from using `Semver` where the semantic version is immutable
to using `ISemver` where the semantic version is a constant. Only update
the L1 contracts, this change should be followed up with additional PRs
that update the other contracts. This change simplifies the code and
reduces the bytecode size of each of the contracts.

tynes committed:
monorepo: run nx repair

Commit the results of running `nx repair` as suggested by nx
during the commit hook.

OptimismBot committed:
Merge pull request #7159 from ethereum-optimism/indexer.bridge.l2.start

fix(indexer): bridge processor fixes

mergify[bot] committed:
Merge branch 'develop' into dependabot/go_modules/op-ufm/github.com/supranational/blst-0.3.11

OptimismBot committed:
Merge pull request #7163 from ethereum-optimism/dependabot/go_modules/op-ufm/github.com/supranational/blst-0.3.11

build(deps): bump github.com/supranational/blst from 0.3.11-0.20230406105308-e9dfc5ee724b to 0.3.11 in /op-ufm

hamdiallam committed:
ci fixes

tynes committed:
ctb: ensure create2 deployer is present

tynes committed:
ctb: fix build

tynes committed:
devnet: refactor

tynes committed:
deps: bump

mergify[bot] committed:
Merge branch 'develop' into dependabot/go_modules/golang.org/x/crypto-0.13.0

tynes committed:
contracts-bedrock: lint

OptimismBot committed:
Merge pull request #7116 from ethereum-optimism/dependabot/go_modules/golang.org/x/crypto-0.13.0

build(deps): bump golang.org/x/crypto from 0.12.0 to 0.13.0

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/chai-and-types/chai-4.3.8

OptimismBot committed:
Merge pull request #7128 from ethereum-optimism/dependabot/npm_and_yarn/chai-and-types/chai-4.3.8

build(deps): bump chai and @types/chai

mergify[bot] committed:
Merge branch 'develop' into dependabot/github_actions/actions/checkout-4

OptimismBot committed:
Merge pull request #7110 from ethereum-optimism/dependabot/github_actions/actions/checkout-4

build(deps): bump actions/checkout from 2 to 4

mergify[bot] committed:
Merge branch 'develop' into dependabot/go_modules/proxyd/github.com/ethereum/go-ethereum-1.12.1

OptimismBot committed:
Merge pull request #7125 from ethereum-optimism/dependabot/go_modules/proxyd/github.com/ethereum/go-ethereum-1.12.1

build(deps): bump github.com/ethereum/go-ethereum from 1.12.0 to 1.12.1 in /proxyd

dependabot[bot] committed:
build(deps): bump github.com/supranational/blst in /proxyd

Bumps [github.com/supranational/blst](https://github.com/supranational/blst) from 0.3.11-0.20230406105308-e9dfc5ee724b to 0.3.11.
- [Release notes](https://github.com/supranational/blst/releases)
- [Commits](https://github.com/supranational/blst/commits/v0.3.11)

---
updated-dependencies:
- dependency-name: github.com/supranational/blst
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

mergify[bot] committed:
Merge branch 'develop' into fix/nx-repair

OptimismBot committed:
Merge pull request #7166 from ethereum-optimism/fix/nx-repair

monorepo: run nx repair

mergify[bot] committed:
Merge branch 'develop' into dependabot/go_modules/proxyd/github.com/supranational/blst-0.3.11

OptimismBot committed:
Merge pull request #7170 from ethereum-optimism/dependabot/go_modules/proxyd/github.com/supranational/blst-0.3.11

build(deps): bump github.com/supranational/blst from 0.3.11-0.20230406105308-e9dfc5ee724b to 0.3.11 in /proxyd

dependabot[bot] committed:
build(deps-dev): bump semver from 5.7.1 to 5.7.2

Bumps [semver](https://github.com/npm/node-semver) from 5.7.1 to 5.7.2.
- [Release notes](https://github.com/npm/node-semver/releases)
- [Changelog](https://github.com/npm/node-semver/blob/v5.7.2/CHANGELOG.md)
- [Commits](https://github.com/npm/node-semver/compare/v5.7.1...v5.7.2)

---
updated-dependencies:
- dependency-name: semver
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

ajsutton committed:
specs: Remove attestation game specs

Add links to specifications from fault proof docs.

hamdiallam committed:
rename some files. comments

ajsutton committed:
op-challenger: Make Position actually immutable

The Position type public API is immutable, but it used mutation internally and used a mix of pointer and non-pointer types. Convert everything to non-pointer types and don't perform any mutations.

OptimismBot committed:
Merge pull request #7174 from ethereum-optimism/aj/position-immutable

op-challenger: Make Position actually immutable

mergify[bot] committed:
Merge branch 'develop' into aj/fpa-docs

tynes committed:
deps: remove dead js deps

The following deps are no longer used in the monorepo:

- lerna
- copyfiles
- mkdirp

This was confirmed using `git grep -rin` for the package
name in the monorepo. Deleting these from the root `package.json`
will reduce some noise in the monorepo by preventing dependabot
update PRs as well as reducing the dependency tree and install time.

tynes committed:
deps: use node lts

Upgrade from node.js v16 to v18. v16 starts end of life on
9/11/2023, meaning it will no longer get security updates.
v18 is currently LTS, https://github.com/nodejs/release#release-schedule
for additional information.

We previously were not able to upgrade node.js versions because
we were using an older version of hardhat that did not support newer
versions of node.js but that is no longer the case.

tynes committed:
chore: add changeset

tynes committed:
contracts-bedrock: semver lock

tynes committed:
op-bindings: fix weth9

tynes committed:
Update ops/docker/Dockerfile.packages

Co-authored-by: Will Cory <willcory10@gmail.com>

ajsutton committed:
op-challenger: Fix make alphabet (#7173)

Root claim needs to have the invalid VM status byte set.

ajsutton committed:
op-challenger: Make solver module work at the entire game level. (#7175)

Introduces GameSolver which evaluates the overall game state and decides all actions to take. Agent now uses GameSolver so that it has no knowledge of the game playing strategy.

mergify[bot] committed:
Merge branch 'develop' into aj/fpa-docs

OptimismBot committed:
Merge pull request #7172 from ethereum-optimism/aj/fpa-docs

specs: Remove attestation game specs

mergify[bot] committed:
Merge branch 'develop' into dependabot/npm_and_yarn/semver-5.7.2

OptimismBot committed:
Merge pull request #7171 from ethereum-optimism/dependabot/npm_and_yarn/semver-5.7.2

build(deps-dev): bump semver from 5.7.1 to 5.7.2

mergify[bot] committed:
Merge branch 'develop' into deps/rm-dead-deps

OptimismBot committed:
Merge pull request #7176 from ethereum-optimism/deps/rm-dead-deps

deps: remove dead js deps

hamdiallam committed:
comments. compute withdrawal hash from the message passer contract for legacy

hamdiallam committed:
nit

mergify[bot] committed:
Merge branch 'develop' into deps/node-lts

OptimismBot committed:
Merge pull request #7178 from ethereum-optimism/deps/node-lts

deps: use node lts

mergify[bot] committed:
Merge branch 'develop' into ctb/create2-deploy

hamdiallam committed:
fix config unit test

OptimismBot committed:
Merge pull request #7164 from ethereum-optimism/ctb/create2-deploy

contracts-bedrock: use create2 to deploy implementations

mergify[bot] committed:
Merge branch 'develop' into indexer.legacy.blocks

OptimismBot committed:
Merge pull request #7119 from ethereum-optimism/indexer.legacy.blocks

feat(indexer): bridge process pre-bedrock blocks

dependabot[bot] committed:
build(deps-dev): bump markdownlint from 0.30.0 to 0.31.0

Bumps [markdownlint](https://github.com/DavidAnson/markdownlint) from 0.30.0 to 0.31.0.
- [Changelog](https://github.com/DavidAnson/markdownlint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/DavidAnson/markdownlint/compare/v0.30.0...v0.31.0)

---
updated-dependencies:
- dependency-name: markdownlint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

tynes committed:
contracts-bedrock: fix semver

Should be a minor bump not patch

OptimismBot committed:
Merge pull request #7189 from ethereum-optimism/dependabot/npm_and_yarn/markdownlint-0.31.0

build(deps-dev): bump markdownlint from 0.30.0 to 0.31.0

tynes committed:
op-bindings: regenerate

tynes committed:
lint: fix

tynes committed:
Merge pull request #7165 from ethereum-optimism/ctb/isemver

contracts-bedrock: migrate to ISemver

lukasrosario committed:
remove permit2 src, add deterministic deployer and devnet permit2 at deterministic address

lukasrosario committed:
remove permit2 from constructors

lukasrosario committed:
untrack devnet config

lukasrosario committed:
remove l1standardbridge function, add comments

lukasrosario committed:
comment on structs
